frontend/src/lib/FileUploadClient.ts

```diff
@@ -56,6 +56,12 @@ export class FileUploadClient extends HttpClient {
       method: "POST",
       data: form,
       onUploadProgress,
+      withCredentials: true,
+      headers: {
+        // This works whether XSRF protection is on or off in the server. If off, this is set to
+        // undefined.
+        "X-XSRFToken": getCookie("_xsrf"),
+      },
```

Comment:
This looks duplicative with the `HttpClient.ts`. I think we should axe these lines and let `HttpClient.ts` handle it.  

As a FYI, setting `withCredentials:true` requires not allowing wildcards `*` for `Access-Control-Allow-Origin`. Right now we only set specific origin if `enableXsrfProtection` is `True` so if this was `False`, the call will fail.

Reply:
I know I know. I was working on the merge :)